File: U1/EXAMEN/01 SERVIDOR/BanquitoServer_Soap_DotNet_GR01/BanquitoServer_Soap_DotNet_GR01/Models/CuotaAmortizacion.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BanquitoServer_Soap_DotNet_GR01.Models
{
    /// <summary>
    /// Entidad CuotaAmortizacion - Representa cada cuota de la tabla de amortización
    /// </summary>
    [Table("CuotaAmortizacion")]
    public class CuotaAmortizacion
    {
        [Key]
        [Column("CuotaId")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long CuotaId { get; set; }

        [Required]
        [Column("NumeroCuota")]
        [Index("IX_CuotaAmortizacion_NumeroCuota")]
        public int NumeroCuota { get; set; }

        [Required]
        [Column("ValorCuota")]
        public decimal ValorCuota { get; set; }

        [Required]
        [Column("Interes")]
        public decimal Interes { get; set; }

        [Required]
        [Column("CapitalPagado")]
        public decimal CapitalPagado { get; set; }

        [Required]
        [Column("Saldo")]
        public decimal Saldo { get; set; }

        // Foreign Key
        [Column("CreditoId")]
        [ForeignKey("Credito")]
        [Index("IX_CuotaAmortizacion_CreditoId")]
        public long CreditoId { get; set; }

        // Propiedad de navegación
        public virtual Credito Credito { get; set; }
    }
}

File: U2/PROYECTO/BanquitoServer_Restfull_Java_GR01/src/main/java/ec/edu/monster/Models/CuotaAmortizacion.java
package ec.edu.monster.Models;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.math.BigDecimal;

/**
 * Entidad Cuota de Amortización
 */
@Entity
@Table(name = "CuotaAmortizacion")
@Data
@NoArgsConstructor
public class CuotaAmortizacion {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "CuotaId")
    private Long cuotaId;

    @Column(name = "NumeroCuota", nullable = false)
    private Integer numeroCuota;

    @Column(name = "ValorCuota", nullable = false, precision = 12, scale = 2)
    private BigDecimal valorCuota;

    @Column(name = "Interes", nullable = false, precision = 12, scale = 2)
    private BigDecimal interes;

    @Column(name = "CapitalPagado", nullable = false, precision = 12, scale = 2)
    private BigDecimal capitalPagado;

    @Column(name = "Saldo", nullable = false, precision = 12, scale = 2)
    private BigDecimal saldo;

    // Foreign Key
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "CreditoId", nullable = false)
    private Credito credito;
}
